packages/gallery/src/components/gallery/proGallery/slideshowView.js

```diff
@@ -208,193 +226,188 @@ class SlideshowView extends GalleryComponent {
         avoidIndividualNavigation,
         ignoreScrollPosition,
         isContinuousScrolling,
+        scrollingUpTheGallery,
       });
     }
     this.removeArrowsIfNeeded();
   }
 
+  getNextItemOrGroupToScrollTo(
+    initiator,
+    direction,
+    ignoreScrollPosition,
+    avoidIndividualNavigation,
+    isAutoTrigger
+  ) {
+     
+    let nextIndex;
+    if (
+      initiator === 'nextGroup' ||
+      (initiator === 'nextItem' &&
+        !ignoreScrollPosition &&
+        avoidIndividualNavigation &&
+        !(this.props.styleParams.groupSize > 1))
+    ) {
+      const key = initiator === 'nextGroup' ? 'groups' : 'galleryItems';
+      nextIndex =
+        this.getCenteredItemOrGroupIdxByScroll(key) + direction;
+    }
+    else if(initiator === 'nextItem') {
+      if(ignoreScrollPosition || !isAutoTrigger) {
+        nextIndex = this.state.activeIndex
+      } else {
+        nextIndex = this.setCurrentItemByScroll();
+      }
+      nextIndex += direction;
+
+      if (!this.props.styleParams.slideshowLoop) {
+        nextIndex = Math.min(
+          this.props.galleryStructure.items.length - 1,
+          nextIndex
+        );
+        nextIndex = Math.max(0, nextIndex);
+      }
+    }
+    return nextIndex;
+  }
+
   async nextItem({
     direction,
     isAutoTrigger,
     scrollDuration,
     avoidIndividualNavigation,
     ignoreScrollPosition,
-    isContinuousScrolling
+    isContinuousScrolling,
+    scrollingUpTheGallery,
   }) {
     if (this.isSliding) {
       return;
     }
     this.isSliding = true;
 
-    let activeIndex;
-    if (ignoreScrollPosition) {
-      activeIndex = this.state.activeIndex;
-    } else {
-      if (
-        avoidIndividualNavigation &&
-        !(this.props.styleParams.groupSize > 1)
-      ) {
-        activeIndex = this.getCenteredItemIdxByScroll();
-      } else {
-        activeIndex = isAutoTrigger
-          ? this.setCurrentItemByScroll()
-          : this.state.activeIndex;
-      }
-    }
-
-    let nextItem = activeIndex + direction;
-    if (!this.props.styleParams.slideshowLoop) {
-      nextItem = Math.min(
-        this.props.galleryStructure.items.length - 1,
-        nextItem
-      );
-      nextItem = Math.max(0, nextItem);
-    }
+    let nextItem = this.getNextItemOrGroupToScrollTo(
+      'nextItem',
+      direction,
+      ignoreScrollPosition,
+      avoidIndividualNavigation,
+      isAutoTrigger
+    );
 
     const { scrollToItem } = this.props.actions;
```

Comment:
`scrollToItem` has no use other than the call to `this.scrollTo`
You can remove it and pass the `this.props.actions.scrollToItem` directly to `this.scrollTo`
Do the same in `nextGroup` as well. There's no need to define an extra `const` if it is used only once.
It makes it harder to see the differences between the two logics.